core/src/mindustry/type/Weapon.java

```diff
@@ -293,9 +293,9 @@ public void update(Unit unit, WeaponMount mount){
 
             mount.reload = reload;
 
-            if(useAmmo){
-                unit.ammo--;
-                if(unit.ammo < 0) unit.ammo = 0;
+            if(useAmmo && !unit.team.rules().infiniteAmmo){
+                unit.ammo -= state.rules.ammoConsumption(unit.team);
+                Math.max(unit.ammo, 0);
```

Comment:
Why was this changed? You're ignoring Math.max's return value.

Reply:
oops, changed it back, if that's what you mean :0